python/pyspark/sql/connect/readwriter.py

```diff
@@ -458,3 +459,61 @@ def orc(
 
     def jdbc(self, *args: Any, **kwargs: Any) -> None:
         raise NotImplementedError("jdbc() not supported for DataFrameWriter")
+
+
+def _test() -> None:
+    import os
+    import sys
+    import doctest
+    from pyspark.sql import SparkSession as PySparkSession
+    from pyspark.testing.connectutils import should_test_connect, connect_requirement_message
+
+    os.chdir(os.environ["SPARK_HOME"])
+
+    if should_test_connect:
+        import pyspark.sql.connect.readwriter
+
+        globs = pyspark.sql.connect.readwriter.__dict__.copy()
+        # Works around to create a regular Spark session
+        sc = SparkContext("local[4]", "sql.connect.readwriter tests", conf=SparkConf())
+        globs["_spark"] = PySparkSession(
+            sc, options={"spark.app.name": "sql.connect.readwriter tests"}
+        )
+
+        # SPARK-41746: infer schema for dictionaries
```

Comment:
I merged this. Mind removing them and rebasing it to see if it passes?

Reply:
done, most passed 👍🏼